stratum/proto/CMakeLists.txt

```diff
@@ -53,112 +22,7 @@ string(JOIN ":" PROTO_IMPORT_PATH
-
-        set(_src ${PB_OUT_DIR}/${_path}/${_name}.pb.cc)
-        set(_hdr ${PB_OUT_DIR}/${_path}/${_name}.pb.h)
-
-        set_source_files_properties(${_src} ${_hdr} PROPERTIES GENERATED TRUE)
-
-        add_custom_command(
-            OUTPUT
-                ${_src} ${_hdr}
-            COMMAND
-                ${PROTOBUF_PROTOC_EXECUTABLE}
-                --proto_path=${PROTO_IMPORT_PATH}
-                --cpp_out=${PB_OUT_DIR}
-                -I${STRATUM_SOURCE_DIR}
-                ${_file}
-            WORKING_DIRECTORY
-                ${CMAKE_CURRENT_SOURCE_DIR}
-            DEPENDS
-                ${SRC_DIR}/${_file}
-            COMMENT
-                "Generating C++ files for ${_file}"
-            VERBATIM
-        )
-
-        # Install .pb.h file in include/.
-        install(
-            FILES
-                ${_hdr}
-            DESTINATION
-                ${PB_HEADER_INSTALL_DIR}/${_path}
-        )
-
-        # Install .proto file in share/.
-        if(INSTALL_PROTO_FILES)
-            install(
-                FILES
-                    ${SRC_DIR}/${_file}
-                DESTINATION
-                    ${DOT_PROTO_INSTALL_DIR}/${_path}
-            )
-        endif()
-    endforeach()
-endfunction(generate_proto_files)
-
-#######################
-# generate_grpc_files #
-#######################
-
-# Generates GRPC C++ files for protobufs.
-function(generate_grpc_files PROTO_FILES SRC_DIR)
-    foreach(_file ${PROTO_FILES})
-        get_filename_component(_path ${_file} DIRECTORY)
-        get_filename_component(_name ${_file} NAME_WE)
-
-        set(_src ${PB_OUT_DIR}/${_path}/${_name}.grpc.pb.cc)
-        set(_hdr ${PB_OUT_DIR}/${_path}/${_name}.grpc.pb.h)
-
-        set_source_files_properties(${_src} ${_hdr} PROPERTIES GENERATED TRUE)
-
-        add_custom_command(
-            OUTPUT
-                ${_src} ${_hdr}
-            COMMAND
-                ${PROTOBUF_PROTOC_EXECUTABLE}
-                --proto_path=${PROTO_IMPORT_PATH}
-                --grpc_out=${PB_OUT_DIR}
-                --plugin=protoc-gen-grpc=${GRPC_CPP_PLUGIN}
-                -I${STRATUM_SOURCE_DIR}
-                ${_file}
-            WORKING_DIRECTORY
-                ${CMAKE_CURRENT_SOURCE_DIR}
-            DEPENDS
-                ${SRC_DIR}/${_file}
-            COMMENT
-                "Generating grpc files for ${_file}"
-            VERBATIM
-        )
-
-        # Install .pb.h file in include/.
-        install(
-            FILES
-                ${_hdr}
-            DESTINATION
-                ${PB_HEADER_INSTALL_DIR}/${_path}
-            )
-
-        # Install .proto file in share/.
-        if(INSTALL_PROTO_FILES)
-            install(
-                FILES
-                    ${SRC_DIR}/${_file}
-                DESTINATION
-                    ${DOT_PROTO_INSTALL_DIR}/${_path}
-            )
-        endif()
-    endforeach()
-endfunction(generate_grpc_files)
+include(ProtobufCompile)
```

Comment:
This is a simple refactoring. I've moved the definitions of the `generate_proto_files()` and `generate_grpc_files() functions to a separate file (`ProtobufCompile.cmake`), reducing the size of `protobuf/CMakeLists.txt` by ~100 lines